java.base/src/main/java/java/net/NetworkInterface$_qbicc.java

```diff
@@ -0,0 +1,537 @@
+/*
+ * This code is based on OpenJDK source file(s) which contain the following copyright notice:
+ *
+ * ------
+ * Copyright (c) 2001, 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ * ------
+ *
+ * This file may contain additional modifications which are Copyright (c) Red Hat and other
+ * contributors.
+ */
+
+package java.net;
+
+import static org.qbicc.runtime.CNative.*;
+import static org.qbicc.runtime.bsd.Ifaddrs.*;
+import static org.qbicc.runtime.bsd.NetIf.*;
+import static org.qbicc.runtime.linux.NetIf.*;
+import static org.qbicc.runtime.posix.ArpaInet.*;
+import static org.qbicc.runtime.posix.Errno.*;
+import static org.qbicc.runtime.posix.SysIoctl.*;
+import static org.qbicc.runtime.posix.NetinetIn.*;
+import static org.qbicc.runtime.posix.NetIf.*;
+import static org.qbicc.runtime.posix.SysSocket.*;
+import static org.qbicc.runtime.posix.Unistd.*;
+import static org.qbicc.runtime.stdc.Errno.*;
+import static org.qbicc.runtime.stdc.Stddef.*;
+import static org.qbicc.runtime.stdc.Stdint.*;
+import static org.qbicc.runtime.stdc.Stdlib.*;
+import static org.qbicc.runtime.stdc.String.*;
+
+import org.qbicc.rt.annotation.Tracking;
+import org.qbicc.runtime.Build;
+
+@Tracking("src/java.base/unix/native/libnet/NetworkInterface.c")
+@Tracking("src/java.base/windows/native/libnet/NetworkInterface.c")
+class NetworkInterface$_qbicc {
+
+    static class netaddr {
+        struct_sockaddr_ptr addr;
```

Comment:
You don't have to change it here, but we should be able to safely move to using e.g. `ptr<struct_sockaddr>` instead now, which might allow us to eventually clean up the proliferate extra subclasses of these object types. If you ever run into a situation where the `ptr<...>` syntax doesn't work, let me know.

Reply:
ok.  I got it backwards.  I thought the T_ptr type was preferred for some reason.  Looks like I'm going to need to do some fixups for Linux, so I could the change to use the ptr<T> uniformly in this file too.

Reply:
I changed for some of the easy cases, but didn't see how to make it work as the argument to a `cast(T_ptr.class)` or a `load(T_ptr.class)`.  Javac didn't like `cast(ptr<T>.class)`.